modules/matterhorn-scheduler-impl/src/main/java/org/opencastproject/scheduler/impl/SchedulerServiceImpl.java

```diff
@@ -1232,6 +1238,32 @@ public int compare(ARecord o1, ARecord o2) {
     return Stream.mk(result).bind(recordToMp).toList();
   }
 
+  /*
+   * Returns a list of conflicting mediapackages.  This method checks for containment, starts-during, and ends-during.
+   */
+  private List<MediaPackage> checkScheduleConflicts(Date checkStart, Date checkEnd, ARecord[] records) {
+    final List<ARecord> result = new ArrayList<>();
+    for (final ARecord r : records) {
+      final Date start = r.getProperties().apply(Properties.getDate(START_DATE_CONFIG));
+      final Date end = r.getProperties().apply(Properties.getDate(END_DATE_CONFIG));
+      if (checkStart.equals(start) || checkEnd.equals(end) ||      //If the start or end times are identical (.after and .before are strict comparisons)
```

Comment:
I think you have missed checkEnd.equals(start) (the potential event ends at the same time when a new event starts) and checkStart.equals(end) (the potential event starts at the same time when a new event finished). Or is this legal?

Reply:
I think there should be a minimum separation enforced between end of one event and start of the next, probably a minute. So ends 9:59am starts 10am is legal, but ends 10am starts 10am is not.

Reply:
@gregorydlogan Could you please add the to corner cases mentioned above and the unit tests for those two cases?

Reply:
@smarquard @staubesv This isn't actually part of the CA API, but I'm not sure that it shouldn't be.  I'll add it to the notes, and to the unit tests.

Reply:
@gregorydlogan Let's assume that our CA API is underspecified: It doesn't make sense to schedule events where Event A starts just 2 seconds after Event B on the same capture agent if that capture agent cannot do this. I think the future CA API should say some words about how fast a capture agent is supposed to be able to stop an recording and start the next one. Probably, it will require at least "some" time to do so... but we would need to ask the capture agent vendor for what is a realistic value here (that can be guaranteed by the CA with reasonable effort)